### ISSUE TITLE
Using CNN-Cert on Keras Application models

### ISSUE THREAD

**Issue by shawnclq** [IBM/CNN-Cert#5]
Hi there, your work is extremely insightful and am hoping to try it out on the models I have.

Does CNN-Cert work on Keras Application models like Xception, VGG16 or ResNet50?
Looking through the code, there are probably some layers like SeparableConv2D which are not in the list of available layers.
How can I include them?

**Comment by AkhilanB** [comment]
Hi,

Thank you for your interest!

CNN-Cert may not directly work on some Keras Application models due to the inclusion of unsupported layers like SeperableConv2D or ZeroPadding2D.
These layers can be included by adding them in the \_\_init\_\_ function of the Model class in cnn_bounds_full.py. In this function, layers like SeperableConv2D can be converted into the equivalent convolution layer and represented by the format of Model (see Dense for an example). Layers like ZeroPadding2D can be included by modifying the previous or next convolution layer (see BatchNormalization for an example).

**Comment by shawnclq** [comment]
Thanks for the prompt reply, I will try out your suggestion!
